### PR TITLE
Language section Vuetified

### PR DESCRIPTION
**Pull request by Pytal**
### To Do

- [x]  Pass initial state from server to client
- [x]  Create service
- [x]  Rewrite php markup and functionality in Vue `apps/settings/templates/settings/personal/persona.info.php`
- [x]  Reload on change
- [x]  Refine input validation
- Rename language strings in other apps (only Talk) to camelCase
    - `activelanguage` → `activeLanguage`
    - `commonlanguages` → `commonLanguages`
    - `languages` → `otherLanguages`
    - [ ]  Talk [https://github.com/nextcloud/spreed/pull/6115](https://github.com/nextcloud/spreed/pull/6115)

### Note: The `TODO` comments are out-of-scope for this PR and will be addressed separately in future PRs

To be merged to `master` after [https://github.com/nextcloud/server/pull/28028](https://github.com/nextcloud/server/pull/28028)

Contributes to #27869

### CODE REVIEW

File: apps/settings/src/components/PersonalInfo/shared/HeaderBar.vue [hunk 1]
```diff
@@ -106,6 +115,21 @@ export default {
 </script>
 
 <style lang="scss" scoped>
+	h3.setting-property {
+		min-height: 37px;
+		position: relative;
+    display: inline-flex;
+    flex-wrap: nowrap;
+    justify-content: flex-start;
+    width: 100%;
```
Comment (artonge) [inline apps/settings/src/components/PersonalInfo/shared/HeaderBar.vue:124]: lint

**Review by artonge — APPROVED** [review]
Looks good

File: apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			<option
+				:value="language.code">
+				{{ language.name }}
+			</option>
+			<option v-for="commonLanguage in commonLanguages"
+				:key="commonLanguage.code"
+				:value="commonLanguage.code">
+				{{ commonLanguage.name }}
+			</option>
+			<optgroup label="––––––––––" />
```
[the remark this answers is not in the document]
Comment (Pytal) [reply apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue]: The option separator, I've updated the markup so it's more semantically clear :)

File: apps/settings/src/components/PersonalInfo/shared/HeaderBar.vue (hunk 1 above)
Comment (Pytal) [reply apps/settings/src/components/PersonalInfo/shared/HeaderBar.vue:124]: I linted it and no output 🤔 these styles were copied from the settings headers that are federation scoped :)

File: apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+				:value="language.code">
+				{{ language.name }}
+			</option>
+			<option v-for="commonLanguage in commonLanguages"
+				:key="commonLanguage.code"
+				:value="commonLanguage.code">
+				{{ commonLanguage.name }}
+			</option>
+			<option disabled>
+				──────────
```
Comment (artonge) [inline apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue:41]: Not sure what is the goal here, but could `<optgroup>` be usefull ? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup

**Review by artonge — APPROVED** [review]

**Review by artonge — CHANGES_REQUESTED** [review]
Oups, misclick on approve

File: apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue (hunk 3 above)
Comment (Pytal) [reply apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue:41]: Separator for common languages and other languages 👇

image

Legacy PHP markup had an empty group for the separator using `optgroup`, this markup does the same thing and is more aligned with what the user sees :)

Comment (artonge) [reply apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue:38]: Hmm, yeah, makes more sens :) thanks

**Review by skjnldsv — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Pull request merged**